mrmustard/physics/ansatze.py

```diff
@@ -417,8 +417,25 @@ def __add__(self, other: PolyExpBase) -> PolyExpBase:
         Adds two ansatze together. This means concatenating them in the batch dimension.
         In the case where c is a polynomial of different shapes it will add padding zeros to make
         the shapes fit. Example: If the shape of c1 is (1,3,4,5) and the shape of c2 is (1,5,4,3) then the
-        shape of the combined object will be (2,5,4,5).
+        shape of the combined object will be (2,5,4,5). It also pads A and b, to account for an eventual
+        different number of polynomial wires.
         """
+        (_, n, _) = self.mat.shape
+        (_, m, _) = other.mat.shape
+        self_num_poly, _ = self.polynomial_shape
```

Comment:
```suggestion
        self_num_poly, a0s = self.polynomial_shape
```
then you can remove `442` below. Ditto for `other_num_poly`.